Library/brew.rb

```diff
@@ -69,15 +69,22 @@ def require?(path)
   # It should never affect external commands so they can handle usage
   # arguments themselves.
 
-  if empty_argv || (help_flag && (cmd.nil? || internal_cmd))
-    # TODO: - `brew help cmd` should display subcommand help
-    require "cmd/help"
-    if empty_argv
-      $stderr.puts ARGV.usage
-    else
+  if empty_argv
+    $stderr.puts ARGV.usage
+    exit 1
+  elsif help_flag
+    if cmd.nil?
       puts ARGV.usage
+      exit 0
+    else
+      # Handle both internal ruby and shell commands
+      require "cmd/help"
+      help_text = Homebrew.help_for_command(cmd)
+      unless help_text.empty?
+        puts help_text
+        exit 0
+      end
```

Comment:
This currently creates a regression for undocumented internal commands, e.g., `brew help --cache` is treated as if `--cache` was an external command, effectively resulting in `brew --cache help`. I think this can be addressed by tweaking the output of `help_for_command` and how it affects what happens here:

- Non-empty string: Documented internal command; show help text and exit.
- Empty string: Undocumented internal command; show something like `No help available for <command>.` followed by generic help and exit.
- `nil`: External command; continue execution and let external command handle the help flag.